lib/src/DateRangePicker/DateRangePickerView.tsx

```diff
@@ -0,0 +1,216 @@
+    Omit<BasePickerProps, 'value' | 'onChange'> {
+  /**
+   * if `true` after selecting `start` date  calendar will not automatically switch to the month of `end` date
+   * @default false
+   */
+  disableAutoMonthSwitching?: boolean;
+}
+
+interface DateRangePickerViewProps
+  extends ExportedDateRangePickerViewProps,
+    CurrentlySelectingRangeEndProps,
+    SharedPickerProps<RangeInput, DateRange> {
+  open: boolean;
+  startText: React.ReactNode;
+  endText: React.ReactNode;
+}
+
+export const DateRangePickerView: React.FC<DateRangePickerViewProps> = ({
+  open,
+  calendars = 2,
+  currentlySelectingRangeEnd,
+  date,
+  disableAutoMonthSwitching = false,
+  disableFuture,
+  disableHighlightToday,
+  disablePast,
+  maxDate: unparsedMaxDate = new Date('2100-01-01'),
+  minDate: unparsedMinDate = new Date('1900-01-01'),
+  onDateChange,
+  onMonthChange,
+  reduceAnimations = defaultReduceAnimations,
+  setCurrentlySelectingRangeEnd,
+  shouldDisableDate,
+  toggleMobileKeyboardView,
+  isMobileKeyboardViewOpen,
+  showToolbar,
+  startText,
+  endText,
+  DateInputProps,
+  ...other
+}) => {
+  const now = useNow();
+  const utils = useUtils();
+  const wrapperVariant = React.useContext(WrapperVariantContext);
+  const minDate = useParsedDate(unparsedMinDate)!;
+  const maxDate = useParsedDate(unparsedMaxDate)!;
+
+  const [start, end] = date;
+  const {
+    changeMonth,
+    calendarState,
+    isDateDisabled,
+    onMonthSwitchingAnimationEnd,
+    changeFocusedDay,
+  } = useCalendarState({
+    date: start || end || now,
+    minDate,
+    maxDate,
+    reduceAnimations,
+    disablePast,
+    disableFuture,
+    onMonthChange,
+    shouldDisableDate,
+    disableSwitchToMonthOnDayFocus: true,
+  });
+
+  const toShowToolbar = showToolbar ?? wrapperVariant !== 'desktop';
+
+  const scrollToDayIfNeeded = (day: MaterialUiPickersDate) => {
+    const displayingMonthRange = wrapperVariant === 'mobile' ? 0 : calendars - 1;
+    const currentMonthNumber = utils.getMonth(calendarState.currentMonth);
+    const requestedMonthNumber = utils.getMonth(day);
+
+    if (
+      requestedMonthNumber < currentMonthNumber ||
+      requestedMonthNumber > currentMonthNumber + displayingMonthRange
+    ) {
+      const newMonth =
+        currentlySelectingRangeEnd === 'start'
+          ? start
+          : // If need to focus end, scroll to the state when "end" is displaying in the last calendar
+            utils.addMonths(end, -displayingMonthRange);
+
+      changeMonth(newMonth);
+    }
+  };
+
+  React.useEffect(() => {
+    if (disableAutoMonthSwitching || !open) {
+      return;
+    }
+
+    if (
+      (currentlySelectingRangeEnd === 'start' && start === null) ||
+      (currentlySelectingRangeEnd === 'end' && end === null)
+    ) {
+      return;
+    }
+
+    scrollToDayIfNeeded(currentlySelectingRangeEnd === 'start' ? start : end);
+  }, [currentlySelectingRangeEnd, date]); // eslint-disable-line
```

Comment:
```suggestion
  }, [currentlySelectingRangeEnd, date]);
```

Reply:
No, because we need to react only on this 2 props change

Reply:
> we need to react only on this 2 props change

I used to believe the same. Experience has shown me that I was wrong. Anytime I tried such an approach, somebody came up with a bug including all the dependencies solves. I would respect if you have reservations about this point. I wouldn't mind actually, I'm confident you will change your mind with the bug reports that will come in :). So, happy to solve that later.

Reply:
This hook is running month switching animation. I suppose nobody will want to run month switching outside of switching month. 

But I agree everything possible. Thanks, let’s wait for reports. 😊

Reply:
I could be wrong too 😁